### DISCUSSION TITLE
Issue with JointJS in Typescript when integrated in own project

### DISCUSSION THREAD

**Discussion by trumpet99** [clientIO/joint#2432] [Q&A]
### What happened?

Followed the sequence for integration of Joint JS typescript project in my own project.
Moved all the necessary files in my project.
Installed all the required packages.
kept the rappid.tgz file in the root folder.

when going for npm start, it is giving the following errors : 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\components'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\services'
 
ERROR in resolving fallback for shared module @clientio/rappid
Module not found: Error: Can't resolve 'file:rappid.tgz' in 'C:\ISF_Code_Checkout\isf_service\Arch_Skeleton_UI_Remote_EDS\eds_components\src\features\workflow\src\shapes'
 
8 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.
 
webpack 5.88.2 compiled with 8 errors in 16000 ms

The folder structure used :  
![image](https://github.com/clientIO/joint/assets/153700615/e84b87a9-89f7-40bb-80f6-3f937e5cc752)

The joint js project is kept in src folder : 
![image](https://github.com/clientIO/joint/assets/153700615/656bfad5-0fd5-4f7a-b1d3-f082f5d064cd)

The package json file is : 
{
  "name": "eds_components",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "webpack serve --config src/config/webpack.dev.js --open --mode development",
    "test": "npx cypress open",
    "build": "webpack --config src/config/webpack.prod.js ",
    "lint": "eslint src/**/*.js --rulesdir src/rules",
    "lint-fix": "eslint src/**/*.js --fix"
  },
  "dependencies": {
    "@apollo/client": "^3.8.8",
    "@azure/msal-browser": "^2.38.0",
    "@azure/msal-react": "^1.5.9",
    "@babel/runtime": "^7.13.10",
    "@clientio/rappid": "file:rappid.tgz",
    "@faker-js/faker": "^8.2.0",
    "@microsoft/applicationinsights-react-js": "^17.0.0",
    "@microsoft/applicationinsights-web": "^3.0.2",
    "@react-gcc-eds/core": "^3.7.0",
    "@tanstack/match-sorter-utils": "^8.8.4",
    "@tanstack/react-table": "^8.10.6",
    "@types/graphlib": "^2.1.12",
    "@types/jest": "^29.5.5",
    "@types/node": "^20.8.4",
    "@types/react": "^18.2.28",
    "@types/react-dom": "^18.2.13",
    "axios": "^1.4.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "dagre": "0.8.4",
    "dhtmlx-gantt": "^7.1.9",
    "dhtmlx-scheduler": "^5.3.13",
    "formBuilder": "^3.9.10",
    "graphlib": "^2.1.8",
    "graphql": "^16.8.1",
    "i18next": "^23.2.3",
    "i18next-browser-languagedetector": "^7.1.0",
    "i18next-http-backend": "^2.2.1",
    "lodash": "^4.17.21",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-i18next": "^13.0.1",
    "react-redux": "^8.1.3",
    "react-router-dom": "^6.14.2",
    "typescript": "^4.5.2",
    "web-vitals": "^3.5.0"
  },
  "devDependencies": {
    "@babel/core": "^7.15.8",
    "@babel/plugin-transform-class-properties": "^7.23.3",
    "@babel/plugin-transform-runtime": "^7.15.8",
    "@babel/preset-env": "^7.15.8",
    "@babel/preset-react": "^7.14.5",
    "@babel/preset-typescript": "^7.10.4",
    "@cypress/code-coverage": "^3.12.10",
    "@tanstack/react-table": "^8.10.7",
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.17.0",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/backbone": "^1.4.18",
    "@types/dagre": "^0.7.42",
    "@types/jest": "23.3.7",
    "@types/jquery": "^3.5.29",
    "@types/node": "10.12.0",
    "@typescript-eslint/eslint-plugin": "^6.9.1",
    "@typescript-eslint/parser": "^6.9.1",
    "autoprefixer": "^10.1.0",
    "babel-loader": "^8.2.2",
    "babel-plugin-istanbul": "^6.1.1",
    "concurrently": "4.1.0",
    "copy-webpack-plugin": "8.1.0",
    "css-loader": "^6.3.0",
    "cypress": "^13.6.0",
    "cypress-istanbul": "^1.3.0",
    "eslint": "^8.51.0",
    "eslint-plugin-react": "^7.33.2",
    "html-webpack-plugin": "^5.3.2",
    "jest": "^29.6.2",
    "jest-html-reporter": "^3.10.1",
    "json-loader": "^0.5.7",
    "mocha": "^10.2.0",
    "nyc": "^15.1.0",
    "react-scripts": "^5.0.1",
    "style-loader": "^3.3.0",
    "ts-loader": "8.1.0",
    "typescript": "3.8.3",
    "webpack": "^5.57.1",
    "webpack-cli": "^4.10.0",
    "webpack-dev-server": "^4.3.1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}

Node modules has the package that shows error : 
![image](https://github.com/clientIO/joint/assets/153700615/d805f3f1-e089-4062-9585-73e86f7fb2e9)

Not able to understand why this issue is there when all the requirements and dependencies are there.

Please provide solution for this!!

### Version

node 20.10.0 (latest)

**Comment by kumilingus** [comment]
Have you tried to remove the `package-lock.json` and `node_modules` and run `npm install` again?

**Reply by trumpet99** [reply]
yes..
 I did that numerous times

**Reply by kumilingus** [reply]
Can you try clearing the cache? i.e. `npm cache clean --force.`
Can you try to downgrade the node version?

**Reply by trumpet99** [reply]
I tried cleaning cache along with downgrading and upgrading the node version... nothing worked in this case

**Reply by kumilingus** [reply]
What is your `webpack` config?
